client_options.go

```diff
@@ -1,70 +1,141 @@
+	var err error
+	if c.accessKey, err = w.initializeAccessKey(); err != nil {
+		c.accessKey = nil
 	}
 }
 
-// WithHTTP will overwrite the default client with a custom client
-func WithHTTP(serverURL string) ClientOps {
-	return func(c *WalletClient) {
-		if c != nil {
-			c.transportOptions = append(c.transportOptions, transports.WithHTTP(serverURL))
-		}
+// adminKeyConf sets the admin key for creating new xpubs
+type adminKeyConf struct {
+	AdminKeyString string
+}
+
+func (w *adminKeyConf) Configure(c *WalletClient) {
+	var err error
+	c.adminXPriv, err = bitcoin.GenerateHDKeyFromString(w.AdminKeyString)
+	if err != nil {
+		c.adminXPriv = nil
 	}
 }
 
-// WithHTTPClient will overwrite the default client with a custom client
-func WithHTTPClient(serverURL string, httpClient *http.Client) ClientOps {
-	return func(c *WalletClient) {
-		if c != nil {
-			c.transportOptions = append(c.transportOptions, transports.WithHTTPClient(serverURL, httpClient))
-		}
+// httpConf sets the URL and httpConf client of a WalletClient
+type httpConf struct {
+	ServerURL  string
+	HTTPClient *http.Client
+}
+
+func (w *httpConf) Configure(c *WalletClient) {
+	// Ensure the ServerURL ends with a clean base URL
+	baseURL, err := validateAndCleanURL(w.ServerURL)
+	if err != nil {
+		// Handle the error appropriately
+		fmt.Println("Invalid URL provided:", err)
+		return
+	}
+
+	const basePath = "/v1"
+	c.server = fmt.Sprintf("%s%s", baseURL, basePath)
+
+	c.httpClient = w.HTTPClient
+	if w.HTTPClient != nil {
+		c.httpClient = w.HTTPClient
+	} else {
+		c.httpClient = http.DefaultClient
 	}
 }
 
-// WithAdminKey will set the admin key for admin requests
-func WithAdminKey(adminKey string) ClientOps {
-	return func(c *WalletClient) {
-		if c != nil {
-			c.transportOptions = append(c.transportOptions, transports.WithAdminKey(adminKey))
+// signRequest configures whether to sign HTTP requests
+type signRequest struct {
+	Sign bool
+}
+
+func (w *signRequest) Configure(c *WalletClient) {
+	c.signRequest = w.Sign
+}
+
+// initializeAccessKey handles the specific initialization of the access key.
+func (w *accessKeyConf) initializeAccessKey() (*bec.PrivateKey, error) {
+	var err error
+	var privateKey *bec.PrivateKey
+	var decodedWIF *wif.WIF
+
+	if decodedWIF, err = wif.DecodeWIF(w.AccessKeyString); err != nil {
+		if privateKey, err = bitcoin.PrivateKeyFromString(w.AccessKeyString); err != nil {
+			return nil, errors.Wrap(err, "failed to decode access key")
 		}
+	} else {
+		privateKey = decodedWIF.PrivKey
 	}
+
+	return privateKey, nil
 }
 
-// WithSignRequest will set whether to sign all requests
-func WithSignRequest(signRequest bool) ClientOps {
-	return func(c *WalletClient) {
-		if c != nil {
-			c.transportOptions = append(c.transportOptions, transports.WithSignRequest(signRequest))
-		}
+// validateAndCleanURL ensures that the provided URL is valid, and strips it down to just the base URL.
+func validateAndCleanURL(rawURL string) (string, error) {
```

Comment:
I think, some unit test would be useful for this function and additionally, it would present how it works (what output is generated by some input)

Reply:
right ....